pkg/v1/cli/pluginmanager/manager.go

```diff
@@ -113,15 +113,30 @@ func DiscoverServerPlugins(serverName string) (plugins []plugin.Discovered, err
 		return
```

Comment:
The returned error here is nil, and I guess that's not intended. This is a consequence of using return vars. 

Generally, you don't need named return vars. Sure, it makes your `return` statements shorter, but that's also a source of confusion for the reader (future you :P). Instead of immediately seeing what's returned you have to traverse back to see what was last assigned to the return var.

Please make your `return` statements explicit. 

And, the tests did *not* catch it, so add a test :)

Reply:
This is intended. If the current server provided is incorrect, I believe we should throw a warning indicating the message instead of throwing an error. (added a warning message)

This will allow users to run `tanzu plugin list`, `tanzu plugin sync` etc commands to sync plugins with a warning message which indicates `Error while discovering contexted plugins`.  I feel we should not block users from running other plugin commands if something is misconfigured (especially someone mistakenly misconfigures tanzu config file)